src/libs/NumberUtils.js

```diff
@@ -0,0 +1,27 @@
+import CONST from '../CONST';
+
+/**
+ * Generates a random positive 64 bit numeric string by randomly generating the left half and right half and concatenating them. Used to generate client-side ids.
+ * @returns {string} randomly generated 64 bit string
+ */
+function rand64() {
+    // Max 64-bit signed:
+    // 9,223,372,036,854,775,807
+    // The left part of the max 64-bit number *+1* because we're flooring it
+    const left = Math.floor(Math.random() * CONST.MAX_64BIT_LEFT_HALF + 1);
```

Comment:
```suggestion
    const left = Math.floor(Math.random() * (CONST.MAX_64BIT_LEFT_HALF + 1));
```

Reply:
We should +1 before multiplying

Reply:
We should actually leave it how it is, or move it the other way right?  Otherwise we can get 0

`Math.floor((Math.random() * CONST.MAX_64BIT_LEFT_HALF) + 1);`

Reply:
Ah that's right! I guess I failed the snake challenge 😅 FWIW Tyler's [original solution](https://expensify.slack.com/archives/C02HWMSMZEC/p1659726344024529?thread_ts=1659032751.555699&cid=C02HWMSMZEC) had the same mistake

Reply:
I edited the suggestion above to `Math.floor((Math.random() * CONST.MAX_64BIT_LEFT_HALF) + 1);`

Reply:
do we need the parentheses? I think it does the same operation regardless but if it helps for clarity I can commit the suggestion.

Reply:
I see this warning:
![Screen Shot 2022-08-19 at 9 48 45 AM](https://user-images.githubusercontent.com/22219519/185657628-c64b653c-8a5a-4ed3-b434-d94431d6b738.png)

Reply:
A result of 0 should be valid here. Particularly for either `left` or `right`, so long as both aren't 0.

```
const left = Math.floor(Math.random() * (CONST.MAX_64BIT_LEFT_HALF + 1));
```

Is correct, it will give a result between 0 and `MAX_64BIT_LEFT_HALF` (9223372036) inclusive of both those numbers. The same pattern is true below for `right`.

If we're concerned about getting `0`, we can check `if (left === 0 and right === 0)` and if so, change `right` to `1`. However, the chance of this happening is orders of magnitude less than this number colliding with another ID in the DB, so we should be able to ignore it, really.